core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockDataMap.java

```diff
@@ -631,6 +638,24 @@ private boolean useMinMaxForExecutorPruning(FilterResolverIntf filterResolverInt
     return useMinMaxForPruning;
   }
 
+  @Override
+  public List<Blocklet> prune(Expression expression, SegmentProperties segmentProperties,
+      List<PartitionSpec> partitions, AbsoluteTableIdentifier identifier) throws IOException {
+    FilterResolverIntf filterResolverIntf = null;
+    if (expression != null) {
+      SegmentProperties properties = getSegmentProperties();
```

Comment:
use the `segmentProperties` object getting passed in method arguments

Reply:
ok